### ISSUE TITLE
Segmented and non-segmented progress bar components throw an axe (accessibility plugin) violation in the newest ruleset

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/vets-design-system-documentation#357]
# Bug Report

- [ ] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

The newest version of the [Deque axe browser plugin](https://deque.com/axe) flags our [Segmented Progressbar](https://design.va.gov/components/progress-bars#segmented-progress-bar) component because it doesn't have an accessible label. This will cause our CI/CD pipeline to break when we [upgrade the axe-core library](https://github.com/department-of-veterans-affairs/va.gov-team/issues/16544). That work is planned for after the holiday.

Pinging @joshkimux for visibility. He's been looking closely at the Progressbar and has some really good insights.

This issue affects our multi-step forms and will block bumping the axe-core library because of the potential for CI/CD build breaks:
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/9226
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/16544

## What I expected to happen

- [ ] Axe reports zero violations of type "ARIA progressbar nodes must have an accessible name"
- [ ] Existing styles are retained

## Reproducing
- Formation version: Latest
- Device: MacBook Pro, MacOS Catalina
- Browser: Chrome latest

Steps to reproduce:

1. Open any multi-step form that users our `<Progressbar />` component
1. Open Chrome / Firefox dev tools
1. Run the axe browser plugin (assumes the plugin has been added to your browser)
1. Verify the serious violation "ARIA progressbar nodes must have an accessible name" appears in your dev tools

## Potential Code Updates
I took a look at how we might update the markup for our Progress bar. Put together a Codepen and the snippet below. JAWS will only read the `aria-valuetext` while NVDA and VoiceOver will read the `aria-label`, then the `aria-valuetext` in preliminary reviews.

* https://codepen.io/tpierce_402/pen/KKgNJLN

```diff
<div>
  <div
!   Consider using a "title" attribute instead, must retest first
+   aria-label="Form 22-1990, education benefits"
    aria-valuemax="8"
    aria-valuemin="0"
    aria-valuenow="1"
+   aria-valuetext="Step 1 of 8: Applicant information"
    class="progress-bar-segmented"
    role="progressbar"
    tabindex="0"
  >
    <div class="progress-segment progress-segment-complete"></div>
    <div class="progress-segment"></div>
    <div class="progress-segment"></div>
    <div class="progress-segment"></div>
    <div class="progress-segment"></div>
    <div class="progress-segment"></div>
    <div class="progress-segment"></div>
    <div class="progress-segment"></div>
  </div>
  <div class="schemaform-chapter-progress">
    <div
-     aria-valuemax="8"
-     aria-valuemin="1"
-     aria-valuenow="1"
-     aria-valuetext="Step 1 of 8: Applicant information"
      class="nav-header nav-header-schemaform"
    >
      <h2 class="vads-u-font-size--h4" id="nav-form-header" tabindex="-1">
        Step 1 of 8: Applicant information
      </h2>
    </div>
  </div>
</div>
```

## Screenshot
![Screen Shot 2020-12-11 at 12 54 04 PM](https://user-images.githubusercontent.com/934879/101958311-b15b9800-3bc8-11eb-8b4e-5192356ef26b.png)

**Comment by caw310** [comment]
@cvalarida @bkjohnson , can you look into this bug next week?

**Comment by JoeTice** [comment]
@caw310 @cvalarida  @bkjohnson  - Have you had an opportunity to look into this? We are trying to determine how we should proceed. Thanks in advance for any info/insights you can provide!

**Comment by caw310** [comment]
@JoeTice , will try to discuss this during our planning on weds and get back to you.

**Comment by cvalarida** [comment]
# Action items for the `SegmetedProgressBar`
- Add a new prop: `ariaValueText`

# Action items for the forms library to use the `ProgressBar`
In `FormNav`:
- Concatenate the `formConfig.subTitle` and `formConfig.title` to pass to the `SegmentedProgressBar` label
- Pass the `stepText` to the progress bar `ariaValueText`

The result will look like the following:
```jsx
<SegmentedProgressBar
  total={chapters.length}
  current={current}
  label={`${formConfig.subTitle}, ${formConfig.title}`}
  ariaValueText={stepText} />
```

# Action items for the ProgressBar
- Figure out how to use it differently
  - Determine where all it's used
  - For each usage, determine what the aria label should be
    - @1Copenut I'll get you the list of uses and will need you to determine what the text should be in each case

**Comment by bkjohnson** [comment]
Showing the error in Storybook (the same appears for the non-segmented version):

![image.png](https://images.zenhubusercontent.com/5db75d6f1484010001787ee1/dc8af071-483f-4c54-9e04-d7f89f340841)

**Comment by bkjohnson** [comment]
Here is how our ProgressBars are used in `vets-website`:

`rg "<(.)*ProgressBar" src/ -g "!test"`


```
src/applications/claims-status/components/UploadStatus.jsx
15:        <ProgressBar percent={progress * 100} />

src/applications/letters/containers/DownloadLetters.jsx
22:      <SegmentedProgressBar total={chapters.length} current={currentStep} />

src/applications/vre/28-1900/orientation/StepComponent.jsx
18:        <SegmentedProgressBar current={data.number} total={stepTotal} />

src/platform/forms-system/src/js/components/FormNav.jsx
83:      <SegmentedProgressBar total={chapters.length} current={current} />

src/platform/forms-system/src/js/fields/FileField.jsx
287:                      <ProgressBar percent={this.state.progress} />
```

**Comment by 1Copenut** [comment]
Thank you @bkjohnson. I'll roll this in with the work I'm doing on the other side for bumping axe-core. Will have answers for you this week hopefully. Lots of moving parts, but we're getting there.

**Comment by 1Copenut** [comment]
### Form Nav text
- **aria-label:**<br/>Form 22-1990, education benefits<br/>_Looks like these are formConfig.subTitle and formConfig.title values_
- **aria-valuetext:**<br/>Step 1 of 8: Applicant information<br/>_Pull out the value from the child `aria-valuetext` and add it here_

**Comment by bkjohnson** [comment]
The Design System team has discussed this, and due to the expanding scope we have decided to pause implementation and hand it off to @JoeTice.

**Comment by 1Copenut** [comment]
This issue has been resolved with [PR #28](https://github.com/department-of-veterans-affairs/component-library/pull/28) in Darius' comment above. Moving to close.

**Comment by joshkimux** [comment]
@1Copenut [Issue 21138 created by Angela](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21138) during her HLR audit may be relevant to this and may require a new ticket

**Comment by 1Copenut** [comment]
@joshkimux I think it will be a new ticket for the design system team, yes. Reading Angela's ticket, I understood that the progressbar shouldn't have a `tabindex=0"` because that made it a focusable element.

Would you mind opening a new ticket on the https://github.com/department-of-veterans-affairs/vets-design-system-documentation/ repo?

**Comment by joshkimux** [comment]
On it!
